src/Shared/ProblemDetails/ProblemDetailsDefaults.cs

```diff
@@ -89,10 +126,17 @@ public static void Apply(ProblemDetails problemDetails, int? statusCode)
             }
         }
 
-        if (Defaults.TryGetValue(problemDetails.Status.Value, out var defaults))
+        var status = problemDetails.Status.GetValueOrDefault();
+        if (Defaults.TryGetValue(status, out var defaults))
         {
             problemDetails.Title ??= defaults.Title;
             problemDetails.Type ??= defaults.Type;
         }
+        else if (problemDetails.Title is null &&
+                 ReasonPhrases.GetReasonPhrase(status) is { } reasonPhrase &&
+                 !string.IsNullOrEmpty(reasonPhrase))
+        {
+            problemDetails.Title = reasonPhrase;
+        }
```

Comment:
_**nit**_: As much as I like the syntax I would prefer, example bellow, that is easier to read:

``` csharp
else if (problemDetails.Title is null) 
{
    var reasonPhrase = ReasonPhrases.GetReasonPhrase(status);
    if (!string.IsNullOrEmpty(reasonPhrase)) 
    {
        problemDetails.Title = reasonPhrase;
    }
}
```